apis/v1alpha1/shared_types.go

```diff
@@ -22,13 +22,25 @@ import (
 )
 
 // InterfaceSelector defines interface to attach to.
-// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MaxProperties=3
```

Comment:
The original intent was to ensure one of PrimaryNodeInterface or Interfaces is set, but not both, and this doesn't do that anymore.  For example, we can now set ExcludeInterfaces, Interfaces, and PrimaryNodeInterface.

Reply:
yeah I spent lots of time trying to come up with cel expersion for this to work but it never worked https://redhat-internal.slack.com/archives/CE4L0F143/p1741606276384279
I was planning to tackle this as we go through the APIs review but to ur point that was the goal allow only one out of the three options